src/App.js

```diff
@@ -1,49 +1,71 @@
-import * as React from "react";
-import { Fragment } from "react";
-import { useState } from 'react';
-import { randomQuote } from "./randomizer.js";
+import React, { useState, useEffect } from 'react';
+import { Quote } from "./Quote.js";
+import { colorizer } from "./colorizer.js";
 
-function Quote({content, author}) {
-  return(
-    <Fragment>
-      <div id="quote">
-        <p id="text">{content}</p>
-        <p id="author">{"—" + author}</p>
-      </div>
-      <span id="tweet-container">
-        <img 
-          id="twitter-logo"
-          src="images/twitter.svg" 
-          alt="Twitter logo" 
-        />
-        <a 
-          id="tweet-quote" 
-          href={"https://twitter.com/intent/tweet?text=“" + content + "” —"+author + ",&via=iggldee"} 
-          target="_blank" 
-          rel="noopener noreferrer" 
-          OnClick="window.open(this.href)" 
-          title="Share on Twitter"
-        >
-          Tweet it<i>!</i>
-        </a>
-      </span>
-    </Fragment>
-  );
-}
+export default function Wrapper() {
+  const [data, setData] = useState(null);
+  const [error, setError] = useState(null);
+  const [customError, setCustomError] = useState(null); // allow for custom-set, more explanatory error messages
+  const [loading, setLoading] = useState(true);
+  const [newQuote, setNewQuote] = useState(true); // establish this var so it can be called later and updated, in turn triggering useEffect's fetch() method to run again upon button click
+  const fetchOptions = {
+    method: 'GET',
+    headers: {
+      'X-RapidAPI-Key': '929124106emshf8e29cf6d861bf0p179527jsn25f292f15772',
+      'X-RapidAPI-Host': 'quotes15.p.rapidapi.com'
+    }
+  };
 
-function New({handleClick}) {
-  return <button id="new-quote" onClick={handleClick}>New quote</button>;
-}
+  // pass quote's length to colorizer() to determine color to associate it to in UI; used by both Quote and New components
+  let color;
+  data && (color = `lab(
+    85% 
+    ${colorizer(data.content.length).a} 
+    ${colorizer(data.content.length).b}
+    )`);
+
+  useEffect(() => {
+    setData(null);
+    setError(null);
+    setLoading(true);
+    const getData = async() => {
+      try {
+        const response = await fetch(
+          'https://quotes15.p.rapidapi.com/quotes/random/', fetchOptions
+        );
+        if (!response.ok) {
+          (response.status === 429) && 
+            (setCustomError("This means the rate of clicks has exceeded what is allowed by the API; try waiting longer before requesting a new quote."));
+          (response.status === 404) &&
+            (setCustomError("This means the API's URL could not be found; if you are the creator of this app, check the URL you are requesting."));
+
+          throw new Error(
+            `An HTTP error has occurred with status ${response.status}`
+          );
+        }
+        let actualData = await response.json();
+        setData(actualData);
```

Comment:
Update the `data` var to have the actual API object we can use; it can be parsed to get the quote and author elsewhere, where `data` var is called